lists/list_cache.go

```diff
@@ -231,35 +247,54 @@ func (b *ListCache) refresh(init bool) error {
 			err = multierror.Append(err, multierror.Prefix(e, fmt.Sprintf("can't create cache group '%s':", group)))
 		}
 
-		if cacheForGroup != nil {
-			b.lock.Lock()
-			b.groupCaches[group] = cacheForGroup
-			b.lock.Unlock()
-		} else {
+		if cacheForGroup == nil {
+			count := b.groupElementCount(group, isInit)
+
 			logger := logger().WithFields(logrus.Fields{
-				"group": group,
+				"group":       group,
+				"total_count": count,
 			})
 
-			if init {
+			if count == 0 {
 				logger.Warn("Populating of group cache failed, cache will be empty until refresh succeeds")
 			} else {
 				logger.Warn("Populating of group cache failed, using existing cache, if any")
 			}
+
+			continue
 		}
 
-		if cacheForGroup != nil {
-			evt.Bus().Publish(evt.BlockingCacheGroupChanged, b.listType, group, cacheForGroup.ElementCount())
+		b.lock.Lock()
+		b.groupCaches[group] = cacheForGroup
+		b.lock.Unlock()
 
-			logger().WithFields(logrus.Fields{
-				"group":       group,
-				"total_count": cacheForGroup.ElementCount(),
-			}).Info("group import finished")
-		}
+		evt.Bus().Publish(evt.BlockingCacheGroupChanged, b.listType, group, cacheForGroup.ElementCount())
+
+		logger().WithFields(logrus.Fields{
+			"group":       group,
+			"total_count": cacheForGroup.ElementCount(),
+		}).Info("group import finished")
 	}
 
 	return err
 }
 
+func (b *ListCache) groupElementCount(group string, isInit bool) int {
```

Comment:
In "Configuration()" we do already count elements per group. Maybe we can reuse it?